### PR TITLE
#302  Slugify tags with theirs groups

### LINKED ISSUE #302: Slugify group names with tags
Now tags slug is calculated only from it's name (value).
But it only group+name pair is unique. So, slugify tags group too

### PR DESCRIPTION
**Pull request by duker33**
Closes #302

### CODE REVIEW

File: catalog/models.py [hunk 1]
```diff
@@ -401,7 +426,7 @@ class Meta:
     def __str__(self):
         return self.name
 
-    def _slugify(self) -> None:
+    def _get_slug(self) -> str:
         """Make a slug from the name."""
```
Comment (artemiy312) [inline catalog/models.py]: This comment is about implementation, that is changing, so let's remove it

File: tests/catalog/test_models.py [hunk 2]
```diff
@@ -266,17 +268,32 @@ def test_long_name(self):
...
         except DataError as e:
             self.assertTrue(False, f'Tag has too long name. {e}')
 
+    # @todo #302:30m  Process more special symbols for slugs.
+    @unittest.expectedFailure
     def test_slugify_conflicts(self):
         slugs = [
             catalog_models.MockTag.objects.create(name=name).slug
-            for name in ['11 A', '1/1 A', '1 1 A']
+            for name in ['11 A', '1/1 A', '1 1 A', '1.1 A', '1-1 A', '1_1 A']
```
Comment (artemiy312) [inline tests/catalog/test_models.py]: Let's create a separate skipped test (e.g. `test_slugify_punctuation_conflicts`) for the set, that led to the conflict to preserve testing of the set that has no such conflicts `11 A, 1/1 A, 1 1 A`

This will help to avoid regressions if we break the slugify for non-conflict set. In common i'd prefer stick to such rule: new tests for new bugs. Up to you

File: tests/catalog/test_models.py [hunk 3]
```diff
@@ -266,17 +268,32 @@ def test_long_name(self):
...
+        ]
+
+        self.assertEqual(len(slugs), len(set(slugs)), msg=slugs)
+
+    def test_slugs_for_cloned_tag_values(self):
```
Comment (artemiy312) [inline tests/catalog/test_models.py]:
```suggestion
    def test_cloned_tag_varied_group_slugs(self):
```
Up to you

Comment (duker33) [reply tests/catalog/test_models.py]: i'll leave it because of few points:
- i want to leave `test_slugs_` at the beginning
- tag values are the same. It's not very common situation, so it's the corner case. That's the point of test problem
- y, groups varying, but it's a normal situation
